drivers/display/display_dummy.c

```diff
@@ -116,19 +112,17 @@ static const struct display_driver_api dummy_display_api = {
 	.set_pixel_format = dummy_display_set_pixel_format,
 };
 
-#define DISPLAY_DUMMY_DEFINE(n)						\
-	static const struct dummy_display_config dd_config_##n = {	\
-		.height = DT_INST_PROP(n, height),			\
-		.width = DT_INST_PROP(n, width),			\
-	};								\
-									\
-	static struct dummy_display_data dd_data_##n;			\
-									\
-	DEVICE_DT_INST_DEFINE(n, &dummy_display_init, NULL,		\
-			      &dd_data_##n,				\
-			      &dd_config_##n,				\
-			      POST_KERNEL,				\
-			      CONFIG_DISPLAY_INIT_PRIORITY,		\
-			      &dummy_display_api);			\
+#define DISPLAY_DUMMY_DEFINE(n)                                                                    \
+	static const struct dummy_display_config dd_config_##n = {                                 \
+		.height = DT_INST_PROP(n, height),                                                 \
+		.width = DT_INST_PROP(n, width),                                                   \
+	};                                                                                         \
+                                                                                                   \
+	static struct dummy_display_data dd_data_##n = {                                           \
+		.current_pixel_format = DT_INST_PROP(n, pixel_format),                             \
+	};                                                                                         \
+                                                                                                   \
+	DEVICE_DT_INST_DEFINE(n, &dummy_display_init, NULL, &dd_data_##n, &dd_config_##n,          \
+			      POST_KERNEL, CONFIG_DISPLAY_INIT_PRIORITY, &dummy_display_api);
```

Comment:
Why cannot "test code" call display_set_pixel_format() and what is wrong with all this lines you change?

Reply:
I'm guessing `clang-format`, @faxe1008 try to add your changes without reformatting the entire block.

Reply:
@jfischer-no the problem is that buffer allocation for the draw buffers happens via a `SYS_INIT` before the tests are run. The pixel format needs to be known beforehand so the appropriate size is allocated.

@pdgendt yep, clang-format. Sure I can fiddel the new field in there, if the non matching slashes are not a concern.

Reply:
> @jfischer-no the problem is that buffer allocation for the draw buffers happens via a `SYS_INIT` before the tests are run. The pixel format needs to be known beforehand so the appropriate size is allocated.

In test, why is it not possible to allocate different buffers for different pixel formats. Why is it not possible to just allocate large enough buffer? Can you use dummy_display_get_capabilities() to get current pixel format?

Reply:
> In test, why is it not possible to allocate different buffers for different pixel formats.
It is possible, but the buffers are not accessible/visible for the tests. See https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/lvgl/lvgl.c#L184 for this. Additionally not only the buffers rely on knowing the pixel format but also the code that sets up the flushing and rounding callbacks:
https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/lvgl/lvgl_display.c#L78
The tests on https://github.com/zephyrproject-rtos/zephyr/pull/68168 are currently failing because via `LV_Z_BITS_PER_PIXEL` the buffer allocation is controlled for the static case but within this code https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/lvgl/lvgl_display.c#L78 the display has its current pixel format set to ARGB8888 (remember this code is run via `SYS_INIT` before any tests). This triggers an internal LVGL assert where the allocated buffer is too small for the supposed ARGB8888 display.

> Why is it not possible to just allocate large enough buffer?
It is possible but in the case of monochrome display with bpp 1 its incredibly wasteful. LVGL is pretty hefty as is, and I want to avoid wasting flash and RAM when possible.

> Can you use dummy_display_get_capabilities() to get current pixel format?
Yes this is precisely what the gluecode does, see my first comment.

Overall - is it really that far fetched of an idea to be able to specify the default pixel format for a display via device tree? At least from my experience the pixel format is not something you mess with during runtime on a regular basis. So why not allow for it to be specified _additionally_ via device tree from the start? Also there are quite a few other bindings that have this.

Reply:
Based on your answer, it seems to be solvable without introducing new bindings. We should also not introduce bindings for testing, and certainly not for testing third party libraries.

> It is possible but in the case of monochrome display with bpp 1 its incredibly wasteful. LVGL is pretty hefty as is, and I want to avoid wasting flash and RAM when possible.

So now it gets very weird, you are modifying dummy display driver. Wasting RAM on what target or platform?

Reply:
> Based on your answer, it seems to be solvable without introducing new bindings.

It is not introducing any new bindings, but a new property to an existing binding.

> We should also not introduce bindings for testing

Ok - but then why do the various `fake-` bindings in various other areas exist?

> So now it gets very weird, you are modifying dummy display driver. Wasting RAM on what target or platform

If you read my previous answer again, I have never said that the new property of the dummy display saves RAM. All I said is that introducing the property allows for properly testing said optimisation.

Reply:
> > Based on your answer, it seems to be solvable without introducing new bindings.
> 
> It is not introducing any new bindings, but a new property to an existing binding.

okay, we should also not introduce new properties for testing.
 
> > We should also not introduce bindings for testing
> 
> Ok - but then why do the various `fake-` bindings in various other areas exist?

No idea, I cannot find any fake- bindings in dts/bindings/display.

> > So now it gets very weird, you are modifying dummy display driver. Wasting RAM on what target or platform
> 
> If you read my previous answer again, I have never said that the new property of the dummy display saves RAM. All I said is that introducing the property allows for properly testing said optimisation.

Let's go back.

>@jfischer-no the problem is that buffer allocation for the draw buffers happens via a SYS_INIT before the tests are run. The pixel format needs to be known beforehand so the appropriate size is allocated.

My question was,

>>Why is it not possible to just allocate large enough buffer?

your answered;

>It is possible but in the case of monochrome display with bpp 1 its incredibly wasteful. LVGL is pretty hefty as is, and I want to avoid wasting flash and RAM when possible.

"Wasting" RAM for dummy display should not be an issue, looks like you need to fix your test instead introducing new property, just allocate large enough buffer, perhaps do not use SYS_INIT.

Please answer the question, "Wasting RAM on what target or platform?"

Reply:
> @jfischer-no the problem is that buffer allocation for the draw buffers happens via a SYS_INIT before the tests are run. The pixel format needs to be known beforehand so the appropriate size is allocated.

I don't have an issue with introducing the binding change, but I'm unclear what SYS_INIT would run before the display init completes- shouldn't the display's device init function complete before `lvgl_init` runs, so this property would be set?